### ISSUE TITLE
E3002: Property IpAddresses should be of type List for resource

### ISSUE THREAD

**Issue by christiangda** [aws-cloudformation/cfn-python-lint#1211]
*cfn-lint version: cfn-lint 0.25.2*

When I use "conditional !If" inside "IpAddresses" properties of "AWS::Route53Resolver::ResolverEndpoint" I get error E3002.

```bash
cfn-lint --info -t vpc-k8s.template 
2019-11-22 10:03:57,241 - cfnlint - INFO - Run scan of template vpc-k8s.template
E3002 Property IpAddresses should be of type List for resource R53ResolverInboundEndPoint at Resources/R53ResolverInboundEndPoint/Properties/IpAddresses/Fn::If/2
vpc-k8s.template:824:11
```
But the template work perfectly and `aws cloudformation validate-template --template-body file://vpc-k8s.template` works fine too

```yaml
  R53ResolverInboundEndPoint:
    Type: AWS::Route53Resolver::ResolverEndpoint
    Condition: CreateR53Resolver
    Properties:
      Direction: INBOUND
      IpAddresses: !If
        - 3AZ
        - - SubnetId: !Ref PrivateSubnet1
          - SubnetId: !Ref PrivateSubnet2
          - SubnetId: !Ref PrivateSubnet3
        - !If
          - 2AZ
          - - SubnetId: !Ref PrivateSubnet1
            - SubnetId: !Ref PrivateSubnet2
          - - "AWS::NoValue"
      Name: !Sub "vpc-${VpcName}-inbound-endpoint"
      SecurityGroupIds:
        - !Ref R53ResolverSecurityGroup
```

![Screenshot 2019-11-22 at 9 39 05 AM](https://user-images.githubusercontent.com/1197820/69412968-35b65100-0d10-11ea-97f3-d71423b63ef0.png)
![Screenshot 2019-11-22 at 9 39 15 AM](https://user-images.githubusercontent.com/1197820/69412981-39e26e80-0d10-11ea-9273-4377975c9d89.png)

**Comment by kddejong** [comment]
Thanks for the details.  Thought we had this covered but obviously not.  I will take a look.
